src/analysis/rust_type.rs

```diff
@@ -65,9 +65,30 @@ fn rust_type_full(env: &Env, type_id: library::TypeId, nullable: Nullable, ref_m
 
         Enumeration(ref enum_) => Ok(enum_.name.clone()),
         Bitfield(ref bitfield) => Ok(bitfield.name.clone()),
-        Record(ref record) => Ok(record.name.clone()),
-        Interface(ref interface) => Ok(interface.name.clone()),
-        Class(ref klass) => Ok(klass.name.clone()),
+        Record(ref record) => {
+            if env.type_status(&type_id.full_name(&env.library)).ignored() {
```

Comment:
If you will collapse matches, remove two TODOs about checking `library::Type::get_name()`